lib/presentation/widget/enter_otp/enter_otp_viewmodel.dart

```diff
@@ -64,24 +68,31 @@ class EnterOTPViewModel extends BaseViewModel {
     _appNavigation.popBack();
   }
 
-  void onLoginPressed(String otp) {
-    store.dispatch(_loginWithOTPAction(otp));
+  void onLoginPressed(String otp, BuildContext context) {
+    store.dispatch(_loginWithOTPAction(otp, context));
   }
 
-  OnlineThunkAction _loginWithOTPAction(String otp) {
+  OnlineThunkAction _loginWithOTPAction(String otp, BuildContext context) {
     return OnlineThunkAction((Store<AppState> store) async {
       await _getPermanentTokenInteractor
           .execute(_enterOTPArgument.baseUrl, _enterOTPArgument.email, _enterOTPArgument.password, otpCode: OTPCode(otp))
           .then(
               (result) => result.fold(
-                  (failure) => {
-                    // todo show toast message here
-                    _appToast.showErrorToast('invalid otp')
-                  },
+                  (failure) => _loginFailure(failure, context),
                   ((success) => _loginSuccess())));
     });
   }
 
+  void _loginFailure(AuthenticationFailure authenticationFailure, BuildContext context) async {
+    var authenException = authenticationFailure.authenticationException;
+    if (authenException is UserLocked) {
+      store.dispatch(AuthenticationAction(Left(authenticationFailure)));
+      await _appNavigation.pushAndRemoveAll(RoutePaths.loginRoute, arguments: AuthenticationArguments(_enterOTPArgument.baseUrl));
```

Comment:
sr, why did u pass `AuthenticationArguments` to arguments here? In case you need arguments for login screen. please create `LoginArguments`

Reply:
yes, it's not necessary, the argument param is redundant.